src/NuGet.Clients/NuGet.PackageManagement.UI/PackageItemLoader.cs

```diff
@@ -202,9 +202,6 @@ public async Task LoadNextAsync(IProgress<IItemLoaderState> progress, Cancellati
         public async Task UpdateStateAsync(IProgress<IItemLoaderState> progress, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
-
-            progress?.Report(_state);
```

Comment:
I deleted this code that reports status to PM UI because of the following two reasons.

- `UpdateStateAsync` invokes `UpdateStateAndReportAsync` method which updates the latest state to PM UI anyways.

- `UpdateStateAsync` is only invoked from `WaitForCompletionAsync` and `WaitForCompletionAsync` methods that run in a loop while waiting for responses from package sources. There is little to no impact because of removing this line of code because the state would have been reported to PM UI in the previous iteration of the `while` loop.

Reply:
Nice catch